source/tests/system/nirfsg_session_tests.cpp

```diff
@@ -0,0 +1,142 @@
+#include <gtest/gtest.h>
+
+#include "device_server.h"
+#include "nirfsg/nirfsg_service.h"
+
+namespace ni {
+namespace tests {
+namespace system {
+
+namespace rfsg = nirfsg_grpc;
+
+const int kInvalidRsrc = -200220;
+const int kInvalidRFSGSession = -1074130544;
+const char* kRFSGErrorResourceNotFoundMessage = "Device identifier is invalid.";
+const char* kInvalidRFSGSessionMessage = "IVI: (Hex 0xBFFA1190) The session handle is not valid.";
+const char* kRFSGTestRsrc = "FakeDevice";
+const char* kRFSGOptionsString = "Simulate=1, DriverSetup=Model:5652; BoardType:PXI";
+const char* kRFSGTestSession = "SessionName";
+const char* kRFSGTestInvalidRsrc = "";
+
+class NiRFSGSessionTest : public ::testing::Test {
+ protected:
+  NiRFSGSessionTest()
+      : device_server_(DeviceServerInterface::Singleton()),
+        nirfsg_stub_(rfsg::NiRFSG::NewStub(device_server_->InProcessChannel()))
+  {
+    device_server_->ResetServer();
+  }
+
+  virtual ~NiRFSGSessionTest() {}
+
+  std::unique_ptr<rfsg::NiRFSG::Stub>& GetStub()
+  {
+    return nirfsg_stub_;
+  }
+
+  ::grpc::Status call_init_with_options(const char* resource_name, const char* option_string, const char* session_name, rfsg::InitWithOptionsResponse* response)
```

Comment:
Are you not using the generated clients because these session tests have some expected grpc-status errors? Or is this just based on the existing session tests?

For the [RFSA tests](https://github.com/astarche/grpc-device/blob/4b56406b7864898850c6ff066bf4a4d161115292/source/tests/system/nirfsa_driver_api_tests.cpp) I did a couple things different:
* Skip the session tests class (just put a simple open/close in the api tests)
* Use the generated client for everything
* Have a very light test class fixture (just managing the stub).

We don't touch the session tests classes much after their written. And it also makes some sense for them to be "more low-level" (though I don't know if that's required). *So I think it's fine to leave this one as-is. *

But I'd like to use the generated client more in the driver api tests to avoid this mass of request building code. That could still have some convenience wrappers for commonly-used-methods-with-uncommonly-used-params (like [this one](https://github.com/ni/grpc-device/blob/361d29b2e906b163ad9e29e04158d5bacb32955d/source/tests/integration/ni_fake_service_tests_endtoend.cpp#L67))

Reply:
Yeah, I was trying to copy another IVI session test class while using the generated client, and somehow I got mixed up and failed. I'll leave this as-is but definitely will use the generated client stuff for the driver API tests. Thanks!